src/main/java/uk/gov/digital/ho/hocs/aws/S3Configuration.java

```diff
@@ -1,43 +1,29 @@
 package uk.gov.digital.ho.hocs.aws;
 
-import com.amazonaws.ClientConfiguration;
 import com.amazonaws.auth.AWSStaticCredentialsProvider;
 import com.amazonaws.auth.BasicAWSCredentials;
 import com.amazonaws.services.s3.AmazonS3;
 import com.amazonaws.services.s3.AmazonS3ClientBuilder;
-import org.springframework.beans.factory.BeanCreationException;
-import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
 import org.springframework.context.annotation.Profile;
-import org.springframework.util.StringUtils;
+import uk.gov.digital.ho.hocs.application.properties.AwsS3Properties;
 
 @Configuration
 @Profile({"s3"})
 public class S3Configuration {
 
-
-    @Bean("s3Client")
-    public static AmazonS3 s3Client(@Value("${document.s3.access-key}") String accessKey,
-                                    @Value("${document.s3.secret-key}") String secretKey,
-                                    @Value("${aws.sqs.region}") String region) {
-
-        if (StringUtils.isEmpty(accessKey)) {
-            throw new BeanCreationException("Failed to create S3 client bean. Need non-blank value for access key");
-        }
-
-        if (StringUtils.isEmpty(secretKey)) {
-            throw new BeanCreationException("Failed to create S3 client bean. Need non-blank values for secret key");
-        }
-
-        if (StringUtils.isEmpty(region)) {
-            throw new BeanCreationException("Failed to create S3 bean. Need non-blank values for region");
-        }
+    @Primary
+    @Bean
+    public static AmazonS3 s3Client(AwsS3Properties awsS3Properties) {
+        var credentialsProvider = new AWSStaticCredentialsProvider(
+                new BasicAWSCredentials(awsS3Properties.getUntrusted().getAccount().getAccessKey(),
```

Comment:
Is there anything you can do to mitigate the number of chain methods going on here?

I know we expect it to work but would surrounding this is a try-catch with appropriate logging etc help if there's an issue?

Reply:
We would get an adequate exception that would highlight where the issue was. If the region is blank for example is says something along the line of 'Region cannot be an empty string'. As this is a bean and initialised at startup we wouldn't have any issues.

Reply:
Also I think you are reviewing an old commit, this has been changed back to just using `@Value`

Reply:
@Value !!!! That is a) Not recommended and b) very poor for SOLID reasons..

In future, we should be using @ConfigurationProperties, @Value is error prone

Reply:
As discussed offline the reasons behind why this changed to `@Value`. I already agreed that I preferred the `@ConfigurationProperties` implementation.